Editor/AGS.Editor/Panes/FreePanControl.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Text;
+using System.Windows.Forms;
+
+namespace AGS.Editor
+{
+    public class FreePanControl : BufferedPanel
+    {
+        private const int WS_EX_COMPOSITED = 0x02000000;
+        private Point _panGrabPoint = Point.Empty;
+        private Cursor _panCursor = Cursors.Hand; // the cursor to use for pan
+        public event EventHandler PanGrabbed;
+        public event EventHandler PanReleased;
+
+        // Enabling Composite makes scrolling smoother
+        protected override CreateParams CreateParams
+        {
+            get
+            {
+                CreateParams cp = base.CreateParams;
+                cp.ExStyle |= WS_EX_COMPOSITED;
+                return cp;
+            }
+        }
+
+        public FreePanControl() : base()
+        {
+
+        }
+
+        protected virtual void OnPanGrab(EventArgs e)
+        {
+            EventHandler handler = PanGrabbed;
```

Comment:
C# trick. Squeeze 3 lines into 1.

```suggestionL35-37
            PanGrabbed?.Invoke(this, EventArgs.Empty);
```

Reply:
I pushed a fix doing this squeeze!